presto-main/src/main/java/com/facebook/presto/sql/ExpressionUtils.java

```diff
@@ -105,11 +106,21 @@ public static Expression binaryExpression(LogicalBinaryExpression.Type type, Col
         requireNonNull(expressions, "expressions is null");
         Preconditions.checkArgument(!expressions.isEmpty(), "expressions is empty");
 
-        // build balanced tree for efficient recursive processing
+        // Build balanced tree for efficient recursive processing.
+        // Preserve the evaluation order of the input expressions.
```

Comment:
This code is pretty non-obvious.  Maybe add a comment explaining what it is doing.